internal/resolver/ooniresolver/ooniresolver.go

```diff
@@ -95,12 +95,28 @@ func (c *Resolver) LookupNS(ctx context.Context, name string) (ns []*net.NS, err
 	return
 }
 
-func (c *Resolver) newQueryWithQuestion(q dns.Question) (query *dns.Msg) {
+// desiredBlockSize is the size that the padded query should me multiple of
+const desiredBlockSize = 128
+
+func (c *Resolver) newQueryWithQuestion(q dns.Question, needspadding bool) (query *dns.Msg) {
 	query = new(dns.Msg)
 	query.Id = dns.Id()
 	query.RecursionDesired = true
 	query.Question = make([]dns.Question, 1)
 	query.Question[0] = q
+	if needspadding {
+		query.SetEdns0(4096, true)
```

Comment:
@xhdix told me we need to better research this value. FTR Google's DoH doesn't accept > 512 octets queries.